src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.List;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public void add(Fruit fruit) {
+        Storage.fruits.add(fruit);
+    }
+
+    @Override
+    public Fruit getFruitByName(String fruitName) {
+        return Storage.fruits.stream()
+                .filter(f -> f.getName().equals(fruitName))
+                .findFirst()
+                .orElseGet(() -> new Fruit(fruitName));
+    }
+
+    @Override
+    public List<Fruit> getAll() {
+        return Storage.fruits;
```

Comment:
Returning the direct reference to the storage list breaks encapsulation. It would be safer to return a new list or an unmodifiable view of the list to prevent external modifications to the internal state of the storage.

Reply:
++